factory/ideas.php

```diff
@@ -52,20 +55,22 @@ class ideas
 	protected $php_ext;
 
 	/**
-	 * @param driver_interface $db
-	 * @param helper           $helper
-	 * @param log              $log
-	 * @param user             $user
-	 * @param string           $table_ideas
-	 * @param string           $table_duplicates
-	 * @param string           $table_rfcs
-	 * @param string           $table_statuses
-	 * @param string           $table_tickets
-	 * @param string           $table_votes
-	 * @param string           $php_ext
+	 * @param \phpbb\config\config              $config
+	 * @param \phpbb\db\driver\driver_interface $db
+	 * @param \phpbb\controller\helper          $helper
+	 * @param \phpbb\log\log                    $log
+	 * @param \phpbb\user                       $user
+	 * @param string                            $table_ideas
+	 * @param string                            $table_duplicates
+	 * @param string                            $table_rfcs
+	 * @param string                            $table_statuses
+	 * @param string                            $table_tickets
+	 * @param string                            $table_votes
+	 * @param string                            $php_ext
 	 */
-	public function __construct(driver_interface $db, helper $helper, log $log, user $user, $table_ideas, $table_duplicates, $table_rfcs, $table_statuses, $table_tickets, $table_votes, $php_ext)
+	public function __construct(\phpbb\config\config $config, \phpbb\db\driver\driver_interface $db, \phpbb\controller\helper $helper, \phpbb\log\log $log, \phpbb\user $user, $table_ideas, $table_duplicates, $table_rfcs, $table_statuses, $table_tickets, $table_votes, $php_ext)
```

Comment:
Because this is just how it is used whereever in the core, but ok, I can just revert that back.